### PR TITLE
Update JCasC version

### PR DESCRIPTION
**Pull request by alecharp**
This is just to update JCasC version to the latest version compatible with the Jenkins Core requirement of GitHub Plugin.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -201,10 +201,9 @@
         </dependency>
 
         <dependency>
-            <groupId>io.jenkins</groupId>
-            <artifactId>configuration-as-code</artifactId>
+            <groupId>io.jenkins.configuration-as-code</groupId>
+            <artifactId>test-harness</artifactId>
             <version>${configuration-as-code.version}</version>
-            <classifier>tests</classifier>
```
Comment (KostyaSha) [inline pom.xml:207]: why should it be hard dep?

Comment (alecharp) [reply pom.xml:207]: I just removed the classifier, this is still a test dependency.

**Comment by KostyaSha** [comment]
Any idea what exactly fails in PR checks?

**Comment by alecharp** [comment]
This is to make sure the plugin is using the latest possible version of CasC to validate that it's still working. Newer version of CasC require a newer version of Jenkins Core.

**Comment by timja** [comment]
this is no longer required, https://github.com/jenkinsci/github-plugin/pull/243 did the update via bom

**Pull request closed without merging**